client/post-editor/controller.js

```diff
@@ -115,6 +115,7 @@ module.exports = {
 
 		function startEditing( siteId ) {
 			context.store.dispatch( setEditorPostId( postID ) );
```

Comment:
>Otherwise we could fire some new `stopEditing` action creator upon unmounting / leaving the editor that has similar effects of resetting edit state.

This is now ringing bells that we'd previously reset edits on a `page.exit` handler but changed away from that approach for some reason. Let me see if I can dig that up.